src/app/vendors/Vendors.tsx

```diff
@@ -121,7 +116,7 @@ function Vendors({
     )
   );
 
-  const onCharacterChanged = (storeId: string) => setCharacterId(storeId);
+  const onCharacterChanged = (storeId: string) => setSelectedStoreId(storeId);
```

Comment:
Could eliminate `onCharacterChanged` or simply assign it to `setSelectedStoreId`